src/main/java/br/com/movile/order/repository/OrderRepository.java

```diff
@@ -1,9 +1,12 @@
 package br.com.movile.order.repository;
 
 import br.com.movile.order.model.Order;
+import org.bson.types.ObjectId;
 import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface OrderRepository extends MongoRepository<Order, String> {
+
+    Order findById(ObjectId orderId);
```

Comment:
Como está o ID da entidade Order? Se for ObjectId então mude o segundo parametro de MongoRepository para ObjectId. Dessa forma, você não precisará criar um findById.

MongoRepository<Order, ObjectId >